contracts/cairo/src/escrow.cairo

```diff
@@ -0,0 +1,251 @@
+use starknet::{ContractAddress, ClassHash, EthAddress};
+
+#[derive(Copy, Drop, Serde, starknet::Store)]
+struct Order {
+    recipient_address: EthAddress,
+    amount: u256,
+    fee: u256
+}
+
+#[starknet::interface]
+trait IEscrow<ContractState> {
+    fn get_order(self: @ContractState, order_id: u256) -> Order;
+
+    fn set_order(ref self: ContractState, order: Order) -> u256;
+
+    fn cancel_order(ref self: ContractState, order_id: u256);
+
+    fn get_order_used(self: @ContractState, order_id: u256) -> bool;
+
+    fn get_order_fee(self: @ContractState, order_id: u256) -> u256;
+
+    // fn withdraw(ref self: ContractState, from_address: felt252, order_id: u256, recipient_address: EthAddress, amount: u256);
+
+    fn get_eth_transfer_contract(self: @ContractState) -> EthAddress;
+    fn get_mm_ethereum_contract(self: @ContractState) -> EthAddress;
+    fn get_mm_starknet_contract(self: @ContractState) -> ContractAddress;
+    
+    fn set_eth_transfer_contract(ref self: ContractState, new_contract: EthAddress);
+    fn set_mm_ethereum_contract(ref self: ContractState, new_contract: EthAddress);
+    fn set_mm_starknet_contract(ref self: ContractState, new_contract: ContractAddress);
+}
+
+#[starknet::contract]
+mod Escrow {
+    use super::{IEscrow, Order};
+
+    use openzeppelin::{
+        access::ownable::OwnableComponent,
+        upgrades::{UpgradeableComponent, interface::IUpgradeable}
+    };
+    use starknet::{
+        ContractAddress, EthAddress, ClassHash, get_caller_address, get_contract_address,
+        get_block_timestamp
+    };
+
+    use yab::interfaces::IERC20::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use yab::interfaces::IEVMFactsRegistry::{
+        IEVMFactsRegistryDispatcher, IEVMFactsRegistryDispatcherTrait
+    };
+
+    /// Components
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
+
+    /// (Ownable)
+    #[abi(embed_v0)]
+    impl OwnableImpl = OwnableComponent::OwnableImpl<ContractState>;
+    impl OwnableInternalImpl = OwnableComponent::InternalImpl<ContractState>;
+
+    /// (Upgradeable)
+    impl InternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
+    // https://github.com/starknet-io/starknet-addresses
+    // MAINNET = GOERLI = GOERLI2
```

Comment:
What is Goerli2? Goerli Layer 2 or sepolia?

Reply:
This should me removed. Goerli 2 was a second testnet starknet had in the past. It was deprecated a couple of weeks ago.